Running multiple build variant license tasks crashes build

In a project with a few different build variants, running a full `gradle build` results in a crash.

The crash seems to be caused by each variant's license task conflicting with one another. When the first license task runs, [it creates a configuration called 'poms'](https://github.com/jaredsburrows/gradle-license-plugin/blob/master/src/main/groovy/com/jaredsburrows/license/LicenseReportTask.groovy#L67). Then when the second variant's license task runs, it attempts to create the same configuration, which crashes since it already exists.

Here's a small example app that demos this issue: https://github.com/egargan/license-plugin-issue-demo.

The error output can be viewed in the  [/gradle-build-error-output](https://github.com/egargan/license-plugin-issue-demo/blob/master/gradle-build-error-output) file. The [/app/build.gradle](https://github.com/egargan/license-plugin-issue-demo/blob/master/app/build.gradle) file sets up the calls to each task.

I get this error:
```
$ gradlew build

> Task :app:licenseDebugReport 
Wrote HTML report to file:///Users/noname/repo/license-plugin-issue-demo/app/build/reports/licenses/licenseDebugReport.html.
Wrote JSON report to file:///Users/noname/repo/license-plugin-issue-demo/app/build/reports/licenses/licenseDebugReport.json.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:licenseReleaseReport'.
> Cannot add a configuration with name 'poms' as a configuration with that name already exists.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
30 actionable tasks: 30 executed
```

--- a/src/main/groovy/com/jaredsburrows/license/LicenseReportTask.groovy
+++ b/src/main/groovy/com/jaredsburrows/license/LicenseReportTask.groovy
@@ -11,14 +11,6 @@ import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.Optional
 
 class LicenseReportTask extends LicenseReportTaskKt {
-  static final def POM_CONFIGURATION = "poms"
-  static final def TEMP_POM_CONFIGURATION = "tempPoms"
-  private static final String ANDROID_SUPPORT_GROUP_ID = "com.android.support"
-  private static final String APACHE_LICENSE_NAME = "The Apache Software License"
-  private static final String APACHE_LICENSE_URL = "http://www.apache.org/licenses/LICENSE-2.0.txt"
-  private static final String OPEN_SOURCE_LICENSES = "open_source_licenses"
-  static final String HTML_EXT = ".html"
-  static final String JSON_EXT = ".json"
   @Optional @Internal def productFlavors = []
 
   /**

--- a/src/main/kotlin/com/jaredsburrows/license/LicenseReportTaskKt.kt
+++ b/src/main/kotlin/com/jaredsburrows/license/LicenseReportTaskKt.kt
@@ -13,15 +13,14 @@ import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 import java.io.File
 import java.net.URI
+import java.util.UUID
 
 abstract class LicenseReportTaskKt : DefaultTask() {
   companion object {
-    private const val POM_CONFIGURATION = "poms"
-    private const val TEMP_POM_CONFIGURATION = "tempPoms"
-    private const val ANDROID_SUPPORT_GROUP_ID = "com.android.support"
-    private const val APACHE_LICENSE_NAME = "The Apache Software License"
-    private const val APACHE_LICENSE_URL = "http://www.apache.org/licenses/LICENSE-2.0.txt"
-    private const val OPEN_SOURCE_LICENSES = "open_source_licenses"
+    const val ANDROID_SUPPORT_GROUP_ID = "com.android.support"
+    const val APACHE_LICENSE_NAME = "The Apache Software License"
+    const val APACHE_LICENSE_URL = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+    const val OPEN_SOURCE_LICENSES = "open_source_licenses"
     const val HTML_EXT = ".html"
     const val JSON_EXT = ".json"
 
@@ -41,6 +40,8 @@ abstract class LicenseReportTaskKt : DefaultTask() {
 //  @Optional @Internal var productFlavors = listOf<com.android.builder.model.ProductFlavor>()
   @OutputFile lateinit var htmlFile: File
   @OutputFile lateinit var jsonFile: File
+  var POM_CONFIGURATION = "poms"
+  var TEMP_POM_CONFIGURATION = "tempPoms"
 
   @TaskAction fun licenseReport() {
     setupEnvironment()
@@ -74,6 +75,9 @@ abstract class LicenseReportTaskKt : DefaultTask() {
    * Setup configurations to collect dependencies.
    */
   private fun setupEnvironment() {
+    POM_CONFIGURATION += POM_CONFIGURATION + variant.orEmpty() + UUID.randomUUID()
+    TEMP_POM_CONFIGURATION += TEMP_POM_CONFIGURATION + variant.orEmpty() + UUID.randomUUID()
+
     // Create temporary configuration in order to store POM information
     project.configurations.apply {
       create(POM_CONFIGURATION)

--- a/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy
+++ b/src/test/groovy/com/jaredsburrows/license/LicensePluginAndroidSpec.groovy
@@ -587,6 +587,70 @@ final class LicensePluginAndroidSpec extends Specification {
                  'licenseFlavor2Flavor4DebugReport', 'licenseFlavor2Flavor4ReleaseReport']
   }
 
+  def 'run build task with buildTypes + productFlavors + flavorDimensions'() {
+    given:
+    def classpathString = pluginClasspath
+      .collect { it.absolutePath.replace('\\', '\\\\') } // escape backslashes in Windows paths
+      .collect { "'$it'" }
+      .join(", ")
+
+    buildFile <<
+      """
+      buildscript {
+        dependencies {
+          classpath files($classpathString)
+        }
+      }
+
+      repositories {
+        maven {
+          url '${mavenRepoUrl}'
+        }
+      }
+
+      apply plugin: 'com.android.application'
+      apply plugin: 'com.jaredsburrows.license'
+
+      android {
+        compileSdkVersion 28
+
+        defaultConfig {
+          applicationId 'com.example'
+        }
+
+        buildTypes {
+          debug {}
+          release {}
+        }
+
+        flavorDimensions 'a', 'b'
+
+        productFlavors {
+          flavor1 { dimension 'a' }
+          flavor2 { dimension 'a' }
+          flavor3 { dimension 'b' }
+          flavor4 { dimension 'b' }
+        }
+      }
+
+      dependencies {
+      }
+      """.stripIndent().trim()
+
+    when:
+    def result = GradleRunner.create()
+      .withProjectDir(testProjectDir.root)
+      .withArguments("licenseFlavor1Flavor3DebugReport", "licenseFlavor1Flavor3ReleaseReport",
+      "licenseFlavor2Flavor4DebugReport", "licenseFlavor2Flavor4ReleaseReport", '-s')
+      .build()
+
+    then:
+    result.task(":licenseFlavor1Flavor3DebugReport").outcome == SUCCESS
+    result.task(":licenseFlavor1Flavor3ReleaseReport").outcome == SUCCESS
+    result.task(":licenseFlavor2Flavor4DebugReport").outcome == SUCCESS
+    result.task(":licenseFlavor2Flavor4ReleaseReport").outcome == SUCCESS
+  }
+
   @Unroll def '#taskName from readme example'() {
     given:
     def classpathString = pluginClasspath

